python/nav/django/auth.py

```diff
@@ -45,10 +47,19 @@
 class AuthenticationMiddleware(MiddlewareMixin):
     def process_request(self, request):
         session = request.session
-
-        if ACCOUNT_ID_VAR not in session:
-            session[ACCOUNT_ID_VAR] = Account.DEFAULT_ACCOUNT
-        account = Account.objects.get(id=session[ACCOUNT_ID_VAR])
+        account = None
+
+        if ACCOUNT_ID_VAR not in session:  # Not logged in
+            if NAV_CONFIG.get('AUTH_SUPPORT_REMOTE_USER', False):
```

Comment:
Do you even need to check the `AUTH_SUPPORT_REMOTE_USER` option here? It seems `authenticate_remote()` will also check this option and return the appropriate response `None` if not enabled.

Reply:
Simplified in "Avoid checking setting twice"